app/parse.py

```diff
@@ -15,9 +33,85 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description"
+        ).text.strip().replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.strip("$")),
+        rating=len(
+            product_soup.find_all("span", class_="ws-icon ws-icon-star")
+        ),
+        num_of_reviews=int(
+            product_soup.select_one(
+                "div.ratings > p.review-count"
+            ).text.split()[0]
+        ),
+    )
+
+
+def load_more(driver: webdriver.Chrome) -> None:
+    while True:
+        try:
+            button = driver.find_element(
+                By.CLASS_NAME,
+                "ecomerce-items-scroll-more"
+            )
+            if button.is_displayed():
+                ActionChains(driver).click(button).perform()
+            else:
+                break
+        except NoSuchElementException:
+            break
+
+
+def get_page_products(driver: webdriver.Chrome, url: str) -> [Product]:
+    driver.get(url)
+    accept_cookies(driver)
+    load_more(driver)
+    soup = BeautifulSoup(driver.page_source, "html.parser")
+    return [
+        parse_single_product(product)
+        for product in soup.select(".thumbnail")
+    ]
+
+
+def accept_cookies(driver: webdriver.Chrome) -> None:
+    try:
```

Comment:
I thin better to create class, for example `ProductParser`
and move these functions inside the class

Reply:
In my opinion, in that case, it's better to use functions because the class can have unnecessary complexity without providing significant benefits.

I am very interested in understanding your perspective better. Could you please elaborate on the advantages of using a class here? 

Thank you!